src/stax/ui_stax.c

```diff
@@ -0,0 +1,310 @@
+        add_word_in_mnemonic(buttonTexts[token - FIRST_SUGGESTION_TOKEN],
+                             strlen(buttonTexts[token - FIRST_SUGGESTION_TOKEN]));
+        if (is_mnemonic_complete()) {
+            display_result_page(check_mnemonic());
+        } else {
+            display_keyboard_page();
+        }
+    }
+}
+
+// function called when a key of keyboard is touched
+static void key_press_callback(const char touchedKey) {
+    size_t textLen = 0;
+    uint32_t mask = 0;
+    const size_t previousTextLen = strlen(textToEnter);
+    if (touchedKey == BACKSPACE_KEY) {
+        if (previousTextLen == 0) {
+            return;
+        }
+        textToEnter[previousTextLen - 1] = '\0';
+        textLen = previousTextLen - 1;
+    } else {
+        textToEnter[previousTextLen] = touchedKey;
+        textToEnter[previousTextLen + 1] = '\0';
+        textLen = previousTextLen + 1;
+    }
+    PRINTF("Current text is: '%s' (size '%d')\n", textToEnter, textLen);
+    if (textLen == 0) {
+        // no suggestion until there is at least 2 characters
+        nbgl_layoutUpdateSuggestionButtons(layout, suggestionIndex, 0, buttonTexts);
+    } else {
+        const size_t nbMatchingWords =
+            bolos_ux_bip39_fill_with_candidates((unsigned char *) &(textToEnter[0]),
+                                                strlen(textToEnter),
+                                                wordCandidates,
+                                                buttonTexts);
+        nbgl_layoutUpdateSuggestionButtons(layout, suggestionIndex, nbMatchingWords, buttonTexts);
+    }
+    if (textLen > 0) {
+        mask = bolos_ux_bip39_get_keyboard_mask((unsigned char *) &(textToEnter[0]),
+                                                strlen(textToEnter));
+    }
+    nbgl_layoutUpdateKeyboard(layout, keyboardIndex, mask, false, LOWER_CASE);
+    nbgl_layoutUpdateEnteredText(layout, textIndex, false, 0, &(textToEnter[0]), false);
+    nbgl_refresh();
+}
+
+static void display_keyboard_page() {
+    nbgl_layoutDescription_t layoutDescription = {.modal = false,
+                                                  .onActionCallback = &keyboard_dispatcher};
+    nbgl_layoutKbd_t kbdInfo = {.lettersOnly = true,   // use only letters
+                                .mode = MODE_LETTERS,  // start in letters mode
+                                .keyMask = 0,          // no inactive key
+                                .callback = &key_press_callback};
+    nbgl_layoutCenteredInfo_t centeredInfo = {.text1 = NULL,
+                                              .text2 = headerText,  // to use as "header"
+                                              .text3 = NULL,
+                                              .style = LARGE_CASE_INFO,
+                                              .icon = NULL,
+                                              .offsetY = 0,
+                                              .onTop = true};
+    strlcpy(textToEnter, "", 1);
```

Comment:
maybe a simpler and more explicit? `textToEnter[0] = '\0'`

Reply:
Better yes